### PR TITLE
feat(feaure-opt-in): settings

### PR DESCRIPTION
**Pull request by corinnekrych**
fixes https://github.com/openshiftio/openshift.io/issues/2402
https://github.com/openshiftio/openshift.io/issues/3295
https://github.com/openshiftio/openshift.io/issues/3254 
https://github.com/openshiftio/openshift.io/issues/3332

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @corinnekrych for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

**Comment by corinnekrych** [comment]
@mindreeper2420 would you mind a quick review of css.
As per Michael Celedonia screens:
https://redhat.invisionapp.com/share/KAF97RBGB#/screens/276638410 in issue
linked in
https://github.com/openshiftio/openshift.io/issues/2402

screenshot 2018-04-26 13 27 02_preview

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.html [hunk 1]
```diff
@@ -1,86 +1,126 @@
...
             </div>
-            <div class="list-view-pf-main-info">
-              <div class="list-view-pf-left">
-                <span class="fa fa-globe list-view-pf-icon-sm"></span>
+          </div>
+        </ng-template>
+
+        <ng-template #untoggledTemplate let-item="item" let-index="index">
+          <div *ngIf="item.expandId === 'released'">
+            <div class="col-sm-1">
```
Comment (mindreeper2420) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.html]: As this and the next `<div>` are `col-sm-*`, but the last is `col-xs-9`, when collapsed to the `xs` bucket, you will have more than 12 columns.

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.html [hunk 2]
```diff
@@ -1,86 +1,126 @@
...
-        </div>
-      </label>
-      <label class="feature-opt-in-label" for="beta">
-        <div class="row">
-          <div class="list-group-item">
-            <div class="list-view-pf-checkbox">
-              <input id="beta" type="radio" name="Beta Features" value="beta" [(ngModel)]="featureLevel" (ngModelChange)="updateProfile()">
+
+          <div *ngIf="item.expandId === 'beta'">
+            <div class="col-sm-1">
```
Comment (mindreeper2420) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.html]: Same thing here - you will have more than 12 columns.

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.html [hunk 3]
```diff
@@ -1,86 +1,126 @@
...
-      </label>
-      <label class="feature-opt-in-label" for="internal">
-        <div class="row">
-          <div class="list-group-item">
-            <div class="list-view-pf-checkbox">
-              <input id="internal" type="radio" name="Internal Features" value="internal" [(ngModel)]="featureLevel" (ngModelChange)="updateProfile()">
+
+          <div *ngIf="item.expandId === 'internal'">
+            <div class="col-sm-1">
+              <span class="fa {{item.icon}} list-pf-icon icon-big"></span>
```
Comment (mindreeper2420) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.html]: instead of using `icon-big` to define the size of the icon, can you use the built-in Font Awesome sizing?

From https://fontawesome.com/v4.7.0/examples/:
`To increase icon sizes relative to their container, use the fa-lg (33% increase), fa-2x, fa-3x, fa-4x, or fa-5x classes.`
`https://github.com/FortAwesome/Font-Awesome/blob/v4.7.0/less/larger.less`

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.less [hunk 4]
```diff
@@ -19,8 +11,32 @@
...
+.feature-details-description {
   margin-bottom: 0;
-  font-weight: normal;
+  font-size: @font-size-small;
+}
+
+.checkbox {
+  border-right: 1px solid @color-pf-black-300;
+  float: left;
+  margin-bottom: 20px;
```
Comment (mindreeper2420) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.less]: update order to
```
margin-top;
margin-right;
margin-bottom;
```

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.less [hunk 5]
```diff
@@ -19,8 +11,32 @@
...
+}
+
+.list-pf-container {
+  padding: 5px 5px 0 30px;
+}
+
+.icon-big {
+  font-size: 2.4em;
+  height: 30px;
+  line-height: 30px;
```
Comment (mindreeper2420) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.less]: line-height shouldn't have a unit of measurement

**Comment by mindreeper2420** [comment]
@corinnekrych - I also noticed that your screenshot differs a little bit from the design, in that your list items have bullets, there isn't any padding around the items in the expanded area and that there is extra spacing between the expansion caret and radio button.

**Comment by corinnekrych** [comment]
@mindreeper2420 mind a second review
@maceledo ^^

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.spec.ts [hunk 6]
```diff
@@ -33,12 +32,22 @@ describe('FeatureOptInComponent', () => {
   let featureOptInComponent: FeatureOptInComponent;
   let gettingStartedServiceMock: jasmine.SpyObj<GettingStartedService>;
   let notificationsMock: jasmine.SpyObj<Notifications>;
-  let userServiceMock: jasmine.SpyObj<UserService>;
+  let userServiceMock: any;
```
Comment (chrislessard) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.spec.ts]: Why did you change the type ?

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.ts [hunk 7]
```diff
@@ -19,20 +17,61 @@ export class FeatureOptInComponent implements OnInit, OnDestroy {
...
+      name: 'released',
+      icon: 'fa-globe',
+      title: 'Production Only Features',
+      description: 'This is the default and current release version of the product.',
+      features: [],
+      displayed: true
+    },
+    {
+      name: 'beta',
+      icon: 'fa-flask',
```
Comment (chrislessard) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.ts]: Since this icon won't be used, why not leave the empty string here ?

**Comment by corinnekrych** [comment]
@chrislessard I've tried as you suggested:
```
    this.subscriptions.push(this.userService.loggedInUser.subscribe(user => {
        if (!this.featureLevel) {
          this.featureLevel = (user.attributes as any).featureLevel;
        }
      }
    ));
``` 
to replace `(this.userService.currentLoggedInUser.attributes as ExtProfile).featureLevel;` but I've noticed that after we updatProfile, the value emitted by the ConnectableObservable `loggedInUser` is stale. 
I'll to stick with my previous proposal (unless you come up with sth that works).

Ty this flow:
1. update your user consent
2. go to another tab on setting
3. go back to opt-in, => make sure the previous selection is still there.

**Comment by fabric8cd** [comment]
@corinnekrych fabric8/fabric8-ui:SNAPSHOT-PR-2796-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2796-fabric8-ui.badger.fabric8.io

**Comment by corinnekrych** [comment]
@mceledonia do you want me to merge this PR:
https://github.com/corinnekrych/fabric8-ui/pull/4
that fixes https://github.com/openshiftio/openshift.io/issues/3332
